File: lanjii/lanjii-modules/module-system/system-biz/src/main/java/com/lanjii/sys/entity/SysMenu.java
package com.lanjii.sys.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lanjii.system.api.dto.SysMenuDTO;
import com.lanjii.system.api.vo.SysMenuVO;
import com.lanjii.framework.mp.base.BaseEntity;
import com.lanjii.sys.base.AdminBaseMapper;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

/**
 * 菜单表(SysMenu)表实体类
 *
 * @author lanjii
 */
@Data
@TableName("sys_menu")
@EqualsAndHashCode(of = "id", callSuper = false)
public class SysMenu extends BaseEntity {

    /**
     * 菜单ID
     */
    private Long id;

    /**
     * 父菜单ID
     */
    private Long parentId;

    /**
     * 祖级列表(逗号分隔)
     */
    private String ancestors;

    /**
     * 菜单名称
     */
    private String name;

    /**
     * 菜单类型（1-目录，2-菜单，3-按钮）
     */
    private Integer type;

    /**
     * 路由路径(前端路由或外链URL)
     */
    private String path;

    /**
     * 组件路径
     */
    private String component;

    /**
     * 权限标识
     */
    private String permission;

    /**
     * 菜单图标
     */
    private String icon;

    /**
     * 显示顺序
     */
    private Integer sortOrder;

    /**
     * 是否可见（0-隐藏，1-显示）
     */
    private Integer isVisible;

    /**
     * 是否启用（1启用 0禁用）
     */
    private Integer isEnabled;

    /**
     * 是否外链（0-否，1-是）
     */
    private Integer isExt;

    /**
     * 打开方式（0-内嵌，1-新窗口）
     */
    private Integer openMode;

    /**
     * 是否页面缓存（0-否，1-是）
     */
    private Integer isKeepAlive;

    /**
     * 显示范围（0-全部，1-租户，2-平台）
     */
    private Integer scope;

    @Mapper
    public interface SysMenuMapper extends AdminBaseMapper<SysMenu, SysMenuVO, SysMenuDTO> {

        @Mapping(target = "children", ignore = true)
        SysMenuVO toVo(SysMenu entity);

        @Mapping(target = "ancestors", ignore = true)
        SysMenu toEntity(SysMenuDTO dto);

    }

    public static final SysMenuMapper INSTANCE = Mappers.getMapper(SysMenuMapper.class);

}

File: lanjii/lanjii-admin-ui/src/types/sys/sysMenu.ts
/** 系统菜单 */
export interface SysMenu {
  /** 菜单ID */
  id?: number
  /** 父菜单ID */
  parentId?: number
  /** 菜单名称 */
  name: string
  /** 菜单类型（1-目录，2-菜单，3-按钮） */
  type: number
  /** 路由路径(前端路由或外链URL) */
  path?: string
  /** 组件路径 */
  component?: string
  /** 权限标识 */
  permission?: string
  /** 菜单图标 */
  icon?: string
  /** 显示顺序 */
  sortOrder?: number
  /** 是否可见（0-隐藏，1-显示） */
  isVisible?: number
  /** 是否启用（1启用 0禁用） */
  isEnabled?: number
  /** 是否外链（0-否，1-是） */
  isExt?: number
  /** 打开方式（0-内嵌，1-新窗口） */
  openMode?: number
  /** 是否页面缓存（0-否，1-是） */
  isKeepAlive?: number
  /** 子菜单列表 */
  children?: SysMenu[]
  /** 创建时间 */
  createTime?: string
  /** 更新时间 */
  updateTime?: string
  /** 创建人 */
  createBy?: string
  /** 更新人 */
  updateBy?: string
  /** 删除标志（0-未删除，1-已删除） */
  deleted?: number
}
